diff -uNr zeromq-4.3.4/tests/testutil.cpp zeromq-4.3.4_patched/tests/testutil.cpp
--- zeromq-4.3.4/tests/testutil.cpp	2021-01-17 13:00:32.000000000 +0100
+++ zeromq-4.3.4_patched/tests/testutil.cpp	2022-04-14 17:20:01.670536873 +0200
@@ -386,7 +386,7 @@
                                protocol_ == IPPROTO_UDP
                                  ? IPPROTO_UDP
                                  : protocol_ == IPPROTO_TCP ? IPPROTO_TCP : 0);
-    TEST_ASSERT_NOT_EQUAL (-1, s_pre);
+    TEST_ASSERT_NOT_EQUAL ((fd_t)-1, s_pre);
 
     if (af_ == AF_INET || af_ == AF_INET6) {
         const char *port = strrchr (endpoint_, ':') + 1;
@@ -443,7 +443,7 @@
                                protocol_ == IPPROTO_UDP
                                  ? IPPROTO_UDP
                                  : protocol_ == IPPROTO_TCP ? IPPROTO_TCP : 0);
-    TEST_ASSERT_NOT_EQUAL (-1, s_pre);
+    TEST_ASSERT_NOT_EQUAL ((fd_t)-1, s_pre);
 
     if (af_ == AF_INET || af_ == AF_INET6) {
 #ifdef ZMQ_HAVE_WINDOWS
